crypto/mem_dbg.c

```diff
@@ -112,7 +112,8 @@ int CRYPTO_mem_ctrl(int mode)
 #else
     int ret = mh_mode;
 
-    CRYPTO_THREAD_run_once(&memdbg_init, do_memdbg_init);
+    if (!RUN_ONCE(&memdbg_init, do_memdbg_init))
+        return 0;
```

Comment:
Although...  considering the gazillion unchecked calls we do on our source, I'm not really looking forward to changing the semantics...